### ISSUE TITLE
 invalid csrf token

### ISSUE THREAD

**Issue by kg3** [NodeBB/NodeBB#5609]
Please include the following information when submitting a bug report/issue:

* NodeBB version and git hash (to find your git hash, execute `git rev-parse HEAD` from the main NodeBB directory)
NodeBB v1.4.5
08f001063c40da2e5d73287fd5c2d9880178602e
* Database (mongo or redis) and it's version.
Redis server v=3.2.8 sha=00000000:0 malloc=jemalloc-4.0.3 bits=64 build=87534321b91909a3
* Exact steps to cause this issue
    1. Sometimes for various users, the logout button does not goto logout. generates errors
    2. On the client I see the error in the console:
        POST https://nodebbwebsite/logout 403 (Forbidden)
    3. On nodebb dev I see the error:
       [23949] - error: /logout
       invalid csrf token
* What you expected
    Clearing cache, site settings, history, relogging in the error would go away
* What happened instead
    When logging in as one user than logging out and logging in as a completely different user makes the error definitely appear.

Thank you!

**Comment by barisusakli** [comment]
Can't seem to repro, is the site being accessed by the same url defined in config.json? Any errors in the browser or server console?

**Comment by kg3** [comment]
Thank you for trying to recreate the error.
Yes the config.json does contain the same url.
The two outputs are directly from the logs you want. The POST error comes from the browser
and the nodebb dev error is the exact output from NodeBB.

What I'm thinking is that since it's not able to be duplicated then it has to do with the serviceworker I have on it. I have ran into a lot more problems having a service worker. I had to restrict the caching to only the *.tpl/*.png files. 

Part of what I can't get to is what ff/chrome do when the service worker is active behind what I can see.  Meaning I think it's probably caching an older csrf-token but I can't see it anywhere in the cache and I don't know where to look for it so I can tell the service worker to leave it alone.

 I know it's not your responsibility to debug a service worker issue so thank you for your help and time :)
